File: ChatGPT(예진)/Python (수정 완료)/Lotto_Program.py
import random

MAX_NUM_SETS = 1000
MAX_NUMBERS = 6
MAX_NUM = 45

# Function: Generate Lotto Numbers
def generate_lotto_numbers():
    chosen = [0] * MAX_NUM
    numbers = []
    i = 0
    while i < MAX_NUMBERS:
        num = random.randint(1, MAX_NUM)
        if not chosen[num - 1]:
            numbers.append(num)
            chosen[num - 1] = 1
            i += 1
    numbers.sort()
    return numbers

# Function: Check Winning Numbers
def check_winner(user_numbers, winning_numbers):
    num_matched = 0
    for num in user_numbers:
        if num in winning_numbers:
            num_matched += 1

    if num_matched == 6:
        return "1st Prize"
    elif num_matched == 5 and winning_numbers[-1] == user_numbers[-1]:
        return "2nd Prize"
    elif num_matched == 5:
        return "3rd Prize"
    elif num_matched == 4:
        return "4th Prize"
    elif num_matched == 3:
        return "5th Prize"
    else:
        return "No Prize"

# Function: Calculate Prize
def calculate_prize(rank):
    if rank[0] == '1':
        return 1000000000
    elif rank[0] == '2':
        return 50000000
    elif rank[0] == '3':
        return 1500000
    elif rank[0] == '4':
        return 50000
    elif rank[0] == '5':
        return 5000
    else:
        return 0

# Main Function
def main():
    num_sets = int(input("Enter the number of Lotto sets to generate: "))
    user_numbers = list(map(int, input("Choose your numbers (6 numbers, separated by space): ").split()))

    winning_numbers_list = [generate_lotto_numbers() for _ in range(num_sets)]

    # Analyze and print numbers
    print("Most frequently drawn numbers: ", end="")
    most_frequent_numbers = {}
    for winning_numbers in winning_numbers_list:
        for number in winning_numbers:
            most_frequent_numbers[number] = most_frequent_numbers.get(number, 0) + 1
    most_frequent = max(most_frequent_numbers, key=most_frequent_numbers.get)
    print(most_frequent, " (", most_frequent_numbers[most_frequent], "times)")

    for i, winning_numbers in enumerate(winning_numbers_list):
        print(f"\nLotto numbers for {i + 1} set: ", end="")
        for number in winning_numbers:
            print(number, end=" ")
        print()
        rank = check_winner(user_numbers, winning_numbers)
        prize = calculate_prize(rank)
        print(f"Winning Result: {rank}, Prize: {prize} won")

if __name__ == "__main__":
    main()

File: ChatGPT(예진)/Java (수정 완료)/Lotto_Program.java
import java.util.*;

public class Lotto_Program {

    private static final int MAX_NUM_SETS = 1000;
    private static final int MAX_NUMBERS = 6;
    private static final int MAX_NUM = 45;

    // Function: Generate Lotto Numbers
    private static List<Integer> generateLottoNumbers() {
        int[] chosen = new int[MAX_NUM];
        List<Integer> numbers = new ArrayList<>();
        int i = 0;
        Random random = new Random();
        while (i < MAX_NUMBERS) {
            int num = random.nextInt(MAX_NUM) + 1;
            if (chosen[num - 1] == 0) {
                numbers.add(num);
                chosen[num - 1] = 1;
                i++;
            }
        }
        Collections.sort(numbers);
        return numbers;
    }

    // Function: Check Winning Numbers
    private static String checkWinner(List<Integer> userNumbers, List<Integer> winningNumbers) {
        int numMatched = 0;
        for (int num : userNumbers) {
            if (winningNumbers.contains(num)) {
                numMatched++;
            }
        }

        if (numMatched == 6) {
            return "1st Prize";
        } else if (numMatched == 5 && winningNumbers.get(MAX_NUMBERS - 1).equals(userNumbers.get(MAX_NUMBERS - 1))) {
            return "2nd Prize";
        } else if (numMatched == 5) {
            return "3rd Prize";
        } else if (numMatched == 4) {
            return "4th Prize";
        } else if (numMatched == 3) {
            return "5th Prize";
        } else {
            return "No Prize";
        }
    }

    // Function: Calculate Prize
    private static int calculatePrize(String rank) {
        if (rank.startsWith("1")) {
            return 1000000000;
        } else if (rank.startsWith("2")) {
            return 50000000;
        } else if (rank.startsWith("3")) {
            return 1500000;
        } else if (rank.startsWith("4")) {
            return 50000;
        } else if (rank.startsWith("5")) {
            return 5000;
        } else {
            return 0;
        }
    }

    // Main Function
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the number of Lotto sets to generate: ");
        int numSets = scanner.nextInt();

        System.out.print("Choose your numbers (6 numbers, separated by space): ");
        List<Integer> userNumbers = new ArrayList<>();
        for (int i = 0; i < MAX_NUMBERS; i++) {
            userNumbers.add(scanner.nextInt());
        }

        List<List<Integer>> winningNumbersList = new ArrayList<>();
        for (int i = 0; i < numSets; i++) {
            winningNumbersList.add(generateLottoNumbers());
        }

        // Analyze and print numbers
        System.out.print("Most frequently drawn numbers: ");
        Map<Integer, Integer> mostFrequentNumbers = new HashMap<>();
        for (List<Integer> winningNumbers : winningNumbersList) {
            for (int number : winningNumbers) {
                mostFrequentNumbers.put(number, mostFrequentNumbers.getOrDefault(number, 0) + 1);
            }
        }
        int mostFrequent = Collections.max(mostFrequentNumbers.entrySet(), Map.Entry.comparingByValue()).getKey();
        System.out.println(mostFrequent + " (" + mostFrequentNumbers.get(mostFrequent) + " times)");

        for (int i = 0; i < winningNumbersList.size(); i++) {
            System.out.print("\nLotto numbers for " + (i + 1) + " set: ");
            for (int number : winningNumbersList.get(i)) {
                System.out.print(number + " ");
            }
            System.out.println();
            String rank = checkWinner(userNumbers, winningNumbersList.get(i));
            int prize = calculatePrize(rank);
            System.out.println("Winning Result: " + rank + ", Prize: " + prize + " won");
        }
    }
}
